### ISSUE TITLE
Unnecessary empty Enum Contructor shows up on report

### ISSUE THREAD

**Issue by theoamonteiro** [jacoco/jacoco#640]
JaCoCo reports missing coverage to enum constructor on empty enum.

### Steps to reproduce
Create a enum like this:
```java
enum EnumExample {
  ;
  public static final String CONST = "This is an empty enum";
}
```
The report shows a Element "EnumExample(String, int)" without any coverage.

JaCoCo version: 0.8.0
Operating system: Linux (Ubuntu 16.04)
Tool integration: Maven
Compiler: javac from OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-0ubuntu0.16.04.2-b12) OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)

### Expected behaviour
In this case the constructor should be ignored and no Element should shown on the report.

### Actual behaviour
The report shows a Element for the enum constructor.

**Comment by Godin** [comment]
Hi @theoamonteiro , thank you for your report. I suppose that forgotten specification of type `java.lang.String` for `CONST` is the only typo?

Also since JaCoCo analyses bytecode and different compilers (ECJ, javac) and their different versions generate different bytecodes, could you please specify used compiler and its version for the completeness of this report?

I also wonder, what is the practical use of an empty enum? This looks like an extreme corner case.

**Comment by theoamonteiro** [comment]
Thanks @Godin (I fixed the typo).

Sometimes instead of
```java
public final class Constants {
    public static final String FOO = "boo";
    private Constants() {
        super();
    }
}
```
You could write
```java
public enum Constants {
   ;
   public static final String FOO = "boo";
}
```
I have to admit that right now I don't remember what was the rational to write an enum instead of a class :-/ but I was thinking that I would not get that Element on the report anymore after version 0.8.0

I don't know if it is a corner case, but since I didn't write that constructor I would expect that it won't show up on my reports, therefore I opened the issue.

**Comment by Godin** [comment]
> was thinking that I would not get that Element on the report anymore after version 0.8.0

This case does not match any of filters implemented in 0.8.0 - see changelog at its usual location at http://www.jacoco.org/jacoco/trunk/doc/changes.html One might think about "private empty no-argument constructors", but in this case constructor has arguments. According to our collection of possible cases for filtering ( https://github.com/jacoco/jacoco/wiki/FilteringOptions ) and AFAIR we've never heard/seen this one before and that's why it looks like a rare corner case to me.

In any case @theoamonteiro thank you for reporting and clarifications.

**Comment by marchof** [comment]
There is a pattern to use enums as a container for static utility methods. Probably those projects are affected.
